### PR TITLE
PB-4812: Fix pod age compare logic

### PR DESCRIPTION
**Pull request by kshithijiyer-px**
**What this PR does / why we need it**:
The current comparePodAge logic breaks because it was dependent on actual name for comparing instead of generated name due to which it will fails as the hash at the end of the  end of pod name like in  `postgres-7fb9569c6b-4mkzv` it is `4mkzv` which fails. 

To fix I am doing the following fixes in this patch: 

- [x] Change logic to use `pod.ObjectMeta.GetGenerateName()` instead of `pod.ObjectMeta.GetName()`
- [x] Use `IsPresent()` to check if namespace in skip list
- [x] Remove redundant kubeconfig switches
- [x] Minor string fixes

Dashboard URL: 
https://aetos.pwx.purestorage.com/resultSet/testSetID/413330 (Latest one with IsPresent)
https://aetos.pwx.purestorage.com/resultSet/testSetID/412342

**Which issue(s) this PR fixes** (optional)
Closes #PB-4812

**Special notes for your reviewer**:

### CODE REVIEW

File: tests/backup/backup_helper.go [hunk 1]
```diff
@@ -5241,20 +5229,21 @@ func comparePodAge(oldPodAge map[string]nsPodAge) error {
...
-			if skipCase == namespace.GetName() {
-				pods, err := k8sCore.GetPods(namespace.ObjectMeta.GetName(), make(map[string]string))
-				if err != nil {
-					return fmt.Errorf("failed to get pods for namespace")
+		pods, err := k8sCore.GetPods(namespace.ObjectMeta.GetName(), make(map[string]string))
+		if err != nil {
+			return fmt.Errorf("failed to get pods for namespace")
+		}
+		log.InfoD("slice utils %v", sliceutil.Contains(namespacesToSkip, namespace.ObjectMeta.GetName()))
+		if sliceutil.Contains(namespacesToSkip, namespace.ObjectMeta.GetName()) {
```
Comment (KrishnaPhalgun28) [inline tests/backup/backup_helper.go:5237]: How about IsPresent?

Comment (kshithijiyer-px) [reply tests/backup/backup_helper.go:5237]: I felt it was better to use inbuil function instead. What's the advantage of using IsPresent() instead?

**Review by sn-px — APPROVED** [review]
LGTM

File: tests/backup/backup_helper.go (hunk 1 above)
Comment (kphalgun-px) [reply tests/backup/backup_helper.go:5237]: I felt there is no need to import a new package.

**Review by KrishnaPhalgun28 — APPROVED** [review]
LGTM

**Review by kphalgun-px — APPROVED** [review]
LGTM

File: tests/backup/backup_helper.go [hunk 2]
```diff
@@ -9,6 +9,7 @@ import (
 	"os/exec"
 	"reflect"
 	"regexp"
+	"sigs.k8s.io/kustomize/kyaml/sliceutil"
```
Comment (mkoppal-px) [inline tests/backup/backup_helper.go:12]: Why is this package needed?

Comment (kshithijiyer-px) [reply tests/backup/backup_helper.go:12]: It's used in line 5236 for checking if a namespace is in a list of namespaces or not

Comment (mkoppal-px) [reply tests/backup/backup_helper.go:12]: We already have a function like this in `backup_helper.go`

```
// IsPresent verifies if the given data is present in slice of data
func IsPresent(dataSlice interface{}, data interface{}) bool {
	s := reflect.ValueOf(dataSlice)
	for i := 0; i < s.Len(); i++ {
		if s.Index(i).Interface() == data {
			return true
		}
	}
	return false
}
```

Comment (kshithijiyer-px) [reply tests/backup/backup_helper.go:12]: What's the benefit of using this over the one which I have used?

Comment (mkoppal-px) [reply tests/backup/backup_helper.go:12]: I do not see any difference between these two functions and it will avoid an import.

Comment (kshithijiyer-px) [reply tests/backup/backup_helper.go]: Done, I still don't understand why we need it.

**Review by mkoppal-px — APPROVED** [review]
LGTM

**Pull request merged**